changelog/@unreleased/pr-4926.v2.yml

```diff
@@ -0,0 +1,8 @@
+type: fix
+fix:
+  description: |-
+    Fix duplicate registration of MetricPublicationFilters. Previously, the same filter could be registered multiple times leading to unnecessary evaluations.
+
+    See internal ticket PG-154380.
```

Comment:
nit: Not convinced we need or want this in the changelog.

Reply:
I'd keep this: it causes a substantial perf regression especially if you use leader blocks (though maybe I should say that)

Reply:
Added a line about it causing perf regressions.

Reply:
I think that's fair. Thanks!